git-ftp

```diff
@@ -1280,6 +1281,16 @@ set_current_branch() {
 	CURRENT_BRANCH="$current"
 }
 
+set_stash() {
+	local stash_config="$(get_config auto-stash)"
+	[ -n "$stash_config" ] && AUTO_STASH="$(boolean $stash_config)"
+	[ $AUTO_STASH -eq 1 ] && git stash push --include-untracked -q -m "Stashed by Git-ftp"
+}
+
+unset_stash(){
+	[ $AUTO_STASH -eq 1 ] && [ $(git stash list -n 1 | grep -q "Stashed by Git-ftp") ] && git stash pop -q
```

Comment:
This will work in most cases but will go wrong from time to time (leading to weird issue reports). Example:

- Working a feature A using auto stash.
- Something goes wrong (network failure) and Git-ftp exits with an error.
- I start debugging / reading docs and give up after a while.
- I start feature B.
- After pushing feature B, my stack from feature A is restored, possibly with conflicts.

A variation of this: I had multiple failures and several Git-ftp stash commits are piled up. They re-surface whenever I have a clean working directory.

Ideally, the stash command would tell us the commit it saved. But it doesn't. We can check the output for `No local changes to save` and skip the pop in that case. Or we extend the stash message with a random token and grep for that. For example:

```
git stash push ... -m "Stashed by Git-ftp. Token: $token"
# ...
[ $(git stash list -n 1 | grep -q "Stashed by Git-ftp. Token: $token") ] && git stash pop
```

But that's probably overcomplicating it.

Reply:
Your first two cases could be handled by just calling  `unset_stash` during `cleanup` which is called before the program exits with an error.

I don’t think it’s that easy to check the output string. Some people have localised versions of git so the output could be in another language and wouldn’t match. But I like the idea with the token. We could just use the current system time (mills) for this. I’ll try this.